include/mount-api-utils.h

```diff
@@ -0,0 +1,192 @@
+#ifndef UTIL_LINUX_MOUNT_API_UTILS
+#define UTIL_LINUX_MOUNT_API_UTILS
+
+#ifdef HAVE_MOUNTFD_API
+
+#include <sys/syscall.h>
+#include <linux/mount.h>
+
+/* Accepted by both open_tree() and mount_setattr(). */
+#ifndef AT_RECURSIVE
+# define AT_RECURSIVE 0x8000
+#endif
+
+#ifndef OPEN_TREE_CLONE
+# define OPEN_TREE_CLONE 1
+#endif
+
+#ifndef OPEN_TREE_CLOEXEC
+# define OPEN_TREE_CLOEXEC O_CLOEXEC
+#endif
+
+#if !defined(HAVE_OPEN_TREE) && defined(SYS_open_tree)
+static inline int open_tree(int dfd, const char *filename, unsigned int flags)
+{
+	return syscall(SYS_open_tree, dfd, filename, flags);
```

Comment:
On error the glibc errors normally return the error value in `errno` while the syscall themselves return it as negative return value.
Maybe we can adapt shims to do this translation too?

Reply:
Good point; that's right that in many places, the library (or applications) uses `return -errno;` (or so), which assumes proper `errno` value. This is probably a generic issue in our syscall fallbacks. It would be nice to improve it.

Any suggestion? -- maybe we can introduce some macros to include/c.h like ul_pre_syscall_errno() and ul_post_syscall_errno(rc) to zeroize and later set the errno.

Reply:
The syscall() function actually does the errno handling.
So my point is moot.